posthog/tasks/test/test_process_scheduled_changes.py

```diff
@@ -0,0 +1,176 @@
+    def test_schedule_feature_flag_set_active(self) -> None:
+        feature_flag = FeatureFlag.objects.create(
+            name="Flag 1",
+            key="flag-1",
+            active=False,
+            filters={"groups": []},
+            team=self.team,
+            created_by=self.user,
+        )
+
+        ScheduledChange.objects.create(
+            team=self.team,
+            record_id=feature_flag.id,
+            model_name="FeatureFlag",
+            payload={"field": "active", "value": True},
+            scheduled_at=(datetime.now(timezone.utc) - timedelta(seconds=30)).isoformat(),
+        )
+
+        process_scheduled_changes()
+
+        updated_flag = FeatureFlag.objects.get(key="flag-1")
+        self.assertEqual(updated_flag.active, True)
+
+    def test_schedule_feature_flag_add_release_condition(self) -> None:
+        feature_flag = FeatureFlag.objects.create(
+            name="Flag 1",
+            key="flag-1",
+            active=False,
+            filters={"groups": []},
+            team=self.team,
+            created_by=self.user,
+        )
+
+        new_release_condition = {
+            "variant": None,
+            "properties": [{"key": "$browser", "type": "person", "value": ["Chrome"], "operator": "exact"}],
+            "rollout_percentage": 30,
+        }
+
+        payload = {
+            "field": "filters",
+            "value": {"groups": [new_release_condition], "payloads": {}, "multivariate": None},
+        }
+
+        ScheduledChange.objects.create(
+            team=self.team,
+            record_id=feature_flag.id,
+            model_name="FeatureFlag",
+            payload=payload,
+            scheduled_at=(datetime.now(timezone.utc) - timedelta(seconds=30)),
+        )
+
+        process_scheduled_changes()
+
+        updated_flag = FeatureFlag.objects.get(key="flag-1")
+        self.assertEqual(updated_flag.filters["groups"][0], new_release_condition)
+
+    def test_schedule_feature_flag_invalid_payload(self) -> None:
+        feature_flag = FeatureFlag.objects.create(
+            name="Flag 1",
+            key="flag-1",
+            active=False,
+            filters={"groups": []},
+            team=self.team,
+            created_by=self.user,
+        )
+
+        payload = {"foo": "bar"}
+
+        scheduled_change = ScheduledChange.objects.create(
+            team=self.team,
+            record_id=feature_flag.id,
+            model_name="FeatureFlag",
+            payload=payload,
+            scheduled_at=(datetime.now(timezone.utc) - timedelta(seconds=30)),
+        )
+
+        process_scheduled_changes()
+
+        updated_flag = FeatureFlag.objects.get(key="flag-1")
+        self.assertEqual(updated_flag.filters["groups"], [])
+
+        updated_scheduled_change = ScheduledChange.objects.get(id=scheduled_change.id)
+        self.assertEqual(updated_scheduled_change.failure_reason, "Invalid payload")
+
+    def test_schedule_feature_flag_multiple_changes(self) -> None:
```

Comment:
can you snapshot the SQL here?

also, not 100% sure this is being tested properly, based on this: 
<img width="907" alt="image" src="https://github.com/PostHog/posthog/assets/7115141/07bf4e5f-509d-4efd-a34d-e49d787f7e59">

I expect this is because there is always a single thread processing the change in all tests, so no concurrency handling is needed.

To simulate production, you can do something like this: https://github.com/PostHog/posthog/blob/master/posthog/test/test_feature_flag_analytics.py#L253 and ensure things are still processed correctly.

Reply:
(as a separate test, no need to change this test, it is good)

Reply:
Will add the test in a follow-up!